### DISCUSSION TITLE
[Feature]: Support `links` in MDX frontmatter

### DISCUSSION THREAD

**Discussion by knowler** [remix-run/remix#2858] [Proposals]
### What is the new or updated feature that you are suggesting?

Currently, `meta` works in MDX frontmatter. It would be great if `links` worked the same way. Following YAML syntax it would look something like this:

```mdx
---
meta:
  title: Welcome to the site
  description: Hello, World!
links:
  - rel: stylesheet
    href: "/welcome.css"
    media: "(prefers-reduced-motion: no-preference)"
---
```

### Why should this feature be included?

While it is likely that Markdown content will share a single stylesheet which can be set in the layout root, sometimes a blog post or page might need to include a stylesheet for something specific included in the content. Currently, there is no way to do this other than include a `<link>` tag in the Markdown.

**Comment by kentcdodds** [comment]
I think this is a good idea. I need to talk with the team about why we have this: https://github.com/remix-run/remix/blob/8e69ed565c3daeb0878c400cf7fe653879645c11/packages/remix-dev/compiler/plugins/mdx.ts#L74

If not for that then you'd be able to do this:

```mdx
---
meta:
  title: Welcome to the site
  description: Hello, World!
---

export const links = () => [
  {
    rel: "stylesheet",
    href: "/welcome.css",
    media: "(prefers-reduced-motion: no-preference)",
  },
]
```

And honestly, what I'd really like to do is make it so you can do `headers` and `meta` like that as well. So we'd support both doing it in frontmatter as well as just doing it within the body of the mdx file. Let me talk to the team about this and I'll let you know whether there's an opportunity for a PR here.

**Reply by kentcdodds** [reply]
Ok, turns out that's there for historical reasons. I've opened this PR to enable exporting links: #3801

**Comment by machour** [comment]
This have been released today in 1.6.6 : https://github.com/remix-run/remix/releases/tag/remix%401.6.6

Enjoy 🎉
